src/grimp/domain/valueobjects.py

```diff
@@ -1,25 +1,14 @@
-from dataclasses import dataclass, astuple
-from typing import Set, Any
+from dataclasses import dataclass
+from typing import Set
 
 
-@dataclass(frozen=True, repr=False, eq=False)
+@dataclass(frozen=True, repr=False)
 class ValueObject:
     def __repr__(self) -> str:
         return "<{}: {}>".format(self.__class__.__name__, self)
 
-    # We use a custom __eq__ method to enforce the liskov principle.
-    # e.g. SpecialModule("foo") == Module("foo") where SpecialModule is a subclass of Module.
-    def __eq__(self, other: Any) -> bool:
-        if isinstance(other, type(self)) or isinstance(self, type(other)):
-            return hash(self) == hash(other)
-        else:
-            return False
-
-    def __hash__(self) -> int:
-        return hash(astuple(self))
```

Comment:
Removing this custom `__eq__` (and `__hash__`) seems a nice change to me. That said, it is a change in behaviour (see removed test `test_equals_obeys_liskov`). So maybe we don't want to change this after all 🤷